tests/unit-tests/ProductDetail.php

```diff
@@ -42,4 +43,92 @@ public function test_view_item_event() {
 		$this->assertEquals( $expected_data, $this->get_event_data(), 'Event data does not match expected data structure for view_item (product_detail()) event' );
 	}
 
+	/**
+	 * Check that `WC_Google_Gtag_JS` registers
+	 * the `assets/js/ga-integration.js` script
+	 * as `woocommerce-google-analytics-integration` (`->script_handle`)
+	 * on `wp_enqueue_scripts` action.
+	 *
+	 * @return void
+	 */
+	public function test_variation_scripts_is_registered() {
+		$gtag = new WC_Google_Gtag_JS();
+
+		// Mimic WC action
+		do_action( 'wp_enqueue_scripts' );
+
+		// Assert the handle is regeistered with the correct name, but not yet enqueued.
+		$this->assertEquals( 'woocommerce-google-analytics-integration', $gtag->script_handle, '`WC_Google_Gtag_JS->script_handle` is not equal `woocommerce-google-analytics-integration`' );
+		$registered_url = wp_scripts()->registered[ $gtag->script_handle ]->src;
+		$this->assertStringContainsString( 'assets/js/ga-integration.js', $registered_url, 'The script does not point to the correct URL' );
+		$this->assertEquals( true, wp_script_is( $gtag->script_handle, 'registered' ), '`woocommerce-google-analytics-integration` script was not registered' );
+		$this->assertEquals( false, wp_script_is( $gtag->script_handle, 'enqueued' ), 'the script is enqueued too early' );
+	}
+
+	/**
+	 * Check that `WC_Google_Gtag_JS` does not enqueue
+	 * the `woocommerce-google-analytics-integration` script
+	 * on `wp_enqueue_scripts` action, for a simple product.
+	 *
+	 * @return void
+	 */
+	public function test_variation_scripts_are_not_enqueued_for_simple() {
+		global $product;
+		$product = WC_Helper_Product::create_simple_product();
+
+		$gtag = new WC_Google_Gtag_JS();
+
+		// Mimic WC action
+		do_action( 'wp_enqueue_scripts' );
+		do_action( 'woocommerce_before_single_product' );
```

Comment:
`do_action( 'woocommerce_before_single_product' );` triggers [woocommerce_ouput_all_notices](https://github.com/woocommerce/woocommerce/blob/3679f019bbe4a1586e28fd9e9d2710e89786f2b1/plugins/woocommerce/includes/wc-template-functions.php#L3840) so although the tests are passing I'm getting some unexpected output when they run:

![Screenshot 2023-05-03 at 06 13 37](https://user-images.githubusercontent.com/40762232/235836942-297def18-ae08-4e02-a10b-1cfa7e43a8e1.png)